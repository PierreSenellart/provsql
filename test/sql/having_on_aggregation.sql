\set ECHO none
\pset format unaligned
SET search_path TO provsql_test,provsql;

CREATE TABLE result_having_on_count AS SELECT
  city,
  COUNT(*),
  sr_formula(provenance(), 'personnel_name') AS formula,
  sr_counting(provenance(),'personnel_count') AS counting
FROM personnel
GROUP BY city
HAVING count(*) > 2 ;

SELECT remove_provenance('result_having_on_count');
SELECT city, formula, counting
FROM result_having_on_count;

DROP TABLE result_having_on_count;

CREATE TABLE result_having_on_sum AS SELECT
  city,
  SUM(id),
  sr_formula(provenance(), 'personnel_name') AS formula,
  sr_counting(provenance(),'personnel_count') AS counting
FROM personnel
GROUP BY city
HAVING SUM(id)>2;

SELECT remove_provenance('result_having_on_sum');
SELECT city, formula, counting
FROM result_having_on_sum;

DROP TABLE result_having_on_sum;
