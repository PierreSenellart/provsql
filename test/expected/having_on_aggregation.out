\set ECHO none
remove_provenance

(1 row)
city|formula|counting
Berlin|(δ(Ellen ⊕ Susan) ⊗ [Ellen*1+Susan*1 > 𝟙*2])|0
New York|(δ(John ⊕ Paul) ⊗ [John*1+Paul*1 > 𝟙*2])|0
Paris|(δ(Dave ⊕ Magdalen ⊕ Nancy) ⊗ [Dave*1+Magdalen*1+Nancy*1 > 𝟙*2])|1
(3 rows)
remove_provenance

(1 row)
city|formula|counting
Berlin|(δ(Ellen ⊕ Susan) ⊗ [Ellen*4+Susan*7 > 𝟙*2])|1
New York|(δ(John ⊕ Paul) ⊗ [John*1+Paul*2 > 𝟙*2])|1
Paris|(δ(Dave ⊕ Magdalen ⊕ Nancy) ⊗ [Dave*3+Magdalen*5+Nancy*6 > 𝟙*2])|1
(3 rows)
